packages/components/src/font-size-picker/utils.ts

```diff
@@ -53,28 +66,28 @@ export function splitValueAndUnitFromSize( size ) {
  * Some themes use css vars for their font sizes, so until we
  * have the way of calculating them don't display them.
  *
- * @param {string|number} value The value that is checked.
- * @return {boolean} Whether the value is a simple css value.
+ * @param  value The value that is checked.
+ * @return Whether the value is a simple css value.
  */
-export function isSimpleCssValue( value ) {
+export function isSimpleCssValue( value: number | string ) {
 	const sizeRegex = /^[\d\.]+(px|em|rem|vw|vh|%)?$/i;
-	return sizeRegex.test( value );
+	return sizeRegex.test( String( value ) );
 }
 
 /**
  * Return font size options in the proper format depending
  * on the currently used control (select, toggle group).
  *
- * @param {boolean}  useSelectControl       Whether to use a select control.
- * @param {Object[]} optionsArray           Array of available font sizes objects.
- * @param {boolean}  disableCustomFontSizes Flag that indicates if custom font sizes are disabled.
- * @return {Object[]|null}                  Array of font sizes in proper format for the used control.
+ * @param  useSelectControl       Whether to use a select control.
+ * @param  optionsArray           Array of available font sizes objects.
+ * @param  disableCustomFontSizes Flag that indicates if custom font sizes are disabled.
+ * @return Array of font sizes in proper format for the used control.
  */
 export function getFontSizeOptions(
-	useSelectControl,
-	optionsArray,
-	disableCustomFontSizes
-) {
```

Comment:
I really wanted to do this! 😅

If you use a conditional type TypeScript doesn't allow you to `return useSelectControl ? ... : ...` because it doesn't know the value of `useSelectControl` at compile-time.

If you use an overload then TypeScript doesn't allow you to call `getFontSizeOptions( shouldUseSelectControl )` ([here](https://github.com/WordPress/gutenberg/blob/b3f0881f5902dc9a2a7b5469ce8c901ae059baee/packages/components/src/font-size-picker/index.tsx#L101)) because it doesn't know the value of `shouldUseSelectControl` at compile-time.

My third idea was to rely on narrowing because although TypeScript doesn't know the value of the boolean at compile-time it, in theory, could infer it based on what branch in `if( shouldUseSelectControl ) { ... } else { ... }` we're in. For this the language gives you [type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) but unfortunately they can only be used as a return type, not a param type.

To be honest I think we should just not use a param here at all and instead have two separate functions. I wanted to limit this PR to only adding types, though, so will do that as a follow-up.

Reply:
I initially thought like writing something like

```jsx
function sampleFunction<T extends boolean>(flag: T): T extends true ? ReturnValueA : ReturnValueB {
  return flag ? 'A' : 'B';
}
```

But with three arguments the function's signature would have become very messy. Two separate functions is much cleaner

> will do that as a follow-up.

Sounds great!